### DISCUSSION TITLE
Help! SemanticTokens/full error

### DISCUSSION THREAD

**Discussion by Yurii-huang** [clangd/clangd#1157] [Q&A] [answered]
` code -32801 ，Request cancelled because the document was modified from: textDocument/semanticTokens/full"`

I notify Clangd by changing one character. When I hit characters many times, it has the following error. 
I check the LSP protocol and write it very broadly.
I don't quite understand how to calculate the normal modification intention.

Thank anyone.

**Comment by sam-mccall** [comment, marked as answer]
You can suppress this behavior by setting the client capability `general.staleRequestSupport.cancel`.

**Reply by Yurii-huang** [reply]
Sorry, I wrote the client myself. It doesn't have similar functions now
In order to better understand the whole protocol and Clangd backend, can you describe the scenario of this error in detail.
Thank U.

**Reply by Yurii-huang** [reply]
It seems that all the usage methods I can understand are not to notify the back-end server according to the change of one character
the client need to implement a timer to trigger with a certain delay, which comes from the senses on other ides.
Or are there other solutions?

**Reply by Yurii-huang** [reply]
`general. staleRequestSupport. cancel
`
I found its usage on the LSP protocol. But this is not the root cause of the error

**Reply by sam-mccall** [reply]
For certain methods, if the content of the file changes before the response is computed, clangd will implicitly cancel the request.
This avoids potentially unbounded queueing for clients that do not cancel stale requests.
If your client reliably cancels stale requests (and ensures the number of outstanding requests is bounded) then it should set the capability in order to disable this behavior of the server.

**Reply by Yurii-huang** [reply]
Can I understand it this way:

I'm changing the file and notifying the server through didchanged. At this time, the server is performing relevant calculation (diagnosis or field analysis), and during this period, I send the didchanged signal again, it will be ignored by the server for usually.
right?

**Reply by Yurii-huang** [reply]
If I can't know the cancellation request correctly.
I can only mask this behavior at the generation entrance of the message.
The scheme triggered by the timer mentioned earlier is the same, is it feasible to do so?
